lib/doorkeeper/config.rb

```diff
@@ -253,9 +243,25 @@ def configure_secrets_for(type, using:, fallback:)
     option :orm,                            default: :active_record
     option :native_redirect_uri,            default: "urn:ietf:wg:oauth:2.0:oob", deprecated: true
     option :grant_flows,                    default: %w[authorization_code client_credentials]
+    option :client_authentication,          default: %w[client_secret_basic client_secret_post none]
```

Comment:
This doesn't actually work as I documented in the PR description, instead it's:

```ruby
Doorkeeper.configure do
  client_authentication %w[client_secret_basic client_secret_post none]
end
```

Instead of
```ruby
Doorkeeper.configure do
  client_authentication :client_secret_basic, :client_secret_post, :none
end
```

With the new `array` flag for `option` we could make it what I thought it was.

Reply:
We probably also need validation that this isn't empty, since that'd cause most of OAuth to fail, I think.